### PR TITLE
Refactor `on_attestation` return signature

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Clean up

**What does this PR do? Why is it needed?**
`on_attestation` returns `([]uint64, error)`. The attesting indices `[]uint64` was never used. This PR removes it

**Which issues(s) does this PR fix?**

Fixes # N/A

**Other notes for review**
N/A

### CODE REVIEW

**Comment by prestonvanloon** [comment]
> beacon-chain/blockchain/receive_attestation.go:39:14: Unhandled error for function call github.com/pkg/errors.Wrap

File: beacon-chain/blockchain/receive_attestation.go [hunk 1]
```diff
@@ -35,9 +35,8 @@ func (s *Service) ReceiveAttestationNoPubsub(ctx context.Context, att *ethpb.Att
 	ctx, span := trace.StartSpan(ctx, "beacon-chain.blockchain.ReceiveAttestationNoPubsub")
 	defer span.End()
 
-	_, err := s.onAttestation(ctx, att)
-	if err != nil {
-		return errors.Wrap(err, "could not process attestation")
+	if err := s.onAttestation(ctx, att); err != nil {
+		errors.Wrap(err, "could not process attestation")
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/receive_attestation.go:39]: ?

Comment (prestonvanloon) [reply beacon-chain/blockchain/receive_attestation.go:39]: Missing return

Comment (terencechain) [reply beacon-chain/blockchain/receive_attestation.go]: Oh man. Copy pasta failed. Surprising Goland didnt alert me for that...

Comment (prestonvanloon) [reply beacon-chain/blockchain/receive_attestation.go]: Thankfully we have static analysis!

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**